hollow/src/main/java/com/netflix/hollow/api/producer/HollowIncrementalProducer.java

```diff
@@ -76,13 +76,22 @@ public void delete(Object obj) {
     public void delete(RecordPrimaryKey key) {
         mutations.put(key, DELETE_RECORD);
     }
-    
+
+    public void clearMutations() {
+        this.mutations.clear();
+    }
+
+    protected ConcurrentHashMap<RecordPrimaryKey, Object> getMutations() {
```

Comment:
I dislike methods that exist just for unit testing. I don't have an alternative suggestion that I like that much more, however.

What about replacing this with a public `boolean hasMutations()` method? I don't think we lose much by changing the unit test from asserting that size == 8 versus just asserting that it's non-empty.

It seems plausible that a `hasMutations()` method might be useful to users. For instance, if you choose to run a cycle on a cadence you could skip cycles for which there haven't been changes (as opposed to running a cycle that ostensibly won't produce a new data state).

Reply:
Agree, I wasn't very comfortable with adding a method just for testing. Your suggestion is great, added the `hasMutations` as public also so it could be useful for users as you mentioned.

Also decided to do 

```
    public boolean hasMutations() { return this.mutations.size() > 0; }
```

instead of 

```
    public boolean hasMutations() { return !this.mutations.isEmpty(); }
```

I'm not a big fan of doing a negative condition... wish Java had `isNotEmpty` like Kotlin or other languages 👎 

Let me know your thoughts

Reply:
@toolbear ^^